okhttp/src/main/java/okhttp3/Request.kt

```diff
@@ -74,11 +64,49 @@ class Request internal constructor(
    * Returns the cache control directives for this response. This is never null, even if this
    * response contains no `Cache-Control` header.
    */
-  fun cacheControl(): CacheControl {
-    return cacheControl ?: CacheControl.parse(headers).also {
-      this.cacheControl = it
+  @get:JvmName("cacheControl") val cacheControl: CacheControl
+    get() {
+      var result = lazyCacheControl
+      if (result == null) {
+        result = CacheControl.parse(headers)
+        lazyCacheControl = result
+      }
+      return result
     }
-  }
+
+  @JvmName("-deprecated_url")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "url"),
+      level = DeprecationLevel.WARNING)
+  fun url(): HttpUrl = url
+
+  @JvmName("-deprecated_method")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "method"),
+      level = DeprecationLevel.WARNING)
+  fun method(): String = method
+
+  @JvmName("-deprecated_headers")
+  @Deprecated(
+      message = "moved to val",
+      replaceWith = ReplaceWith(expression = "headers"),
+      level = DeprecationLevel.WARNING)
+  fun headers(): Headers = headers
+  @JvmName("-deprecated_body")
```

Comment:
super nit: should have empty line above this

Reply:
Thanks! fixed it! :)